includes/Core/Conversion_Tracking/Conversion_Tracking.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Conversion_Tracking
+ *
+ * @package   Google\Site_Kit\Core\Modules
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Conversion_Tracking;
+
+use Google\Site_Kit\Context;
+use InvalidArgumentException;
+
+/**
+ * Class for managing conversion tracking.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Conversion_Tracking {
+
+	/**
+	 * Context object.
+	 *
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * Supported conversion event providers.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	public static $conversion_event_providers = array();
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context $context Plugin context.
+	 */
+	public function __construct( Context $context ) {
+		$this->context = $context;
+	}
+
+	/**
+	 * Registers the class functionality.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		$active_conversion_event_providers = $this->get_active_conversion_event_providers();
+
+		array_walk(
+			$active_conversion_event_providers,
+			function( Conversion_Events_Provider $active_conversion_event_provider ) {
+				$script_asset = $active_conversion_event_provider->register_script();
+				$script_asset->enqueue();
+			}
+		);
+	}
+
+	/**
+	 * Gets the instances of active conversion event providers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of active Conversion_Events_Provider instances.
+	 */
+	public function get_active_conversion_event_providers() {
+		$active_conversion_event_providers = array();
+
+		foreach ( self::$conversion_event_providers as $conversion_event_provider_class ) {
+			$instance = new $conversion_event_provider_class( $this->context );
+
+			if ( $instance->is_active() ) {
+				$active_conversion_event_providers[ $instance::CONVERSION_EVENT_PROVIDER_SLUG ] = $instance;
+			}
+		}
```

Comment:
The conversion_event_providers list is supposed to contain key/value pairs where keys are slugs and values are class names. It means that we need to use slugs that come from keys instead of relying on the `CONVERSION_EVENT_PROVIDER_SLUG` constant that may exist in the provider class or may not. Please, update this loop to use keys as slugs for providers.

Reply:
Ah indeed, thanks